### PR TITLE
Use pyproject.toml

### PR DESCRIPTION
**Pull request by ayalash**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
-
-.env/.up-to-date: setup.py Makefile
-	python -m venv .env
-	.env/bin/pip install -e ".[testing]"
-	touch $@
+	uv run --extra testing pytest tests
 
+env:
+	uv venv
+	uv pip install -e ".[testing]"
```
Comment (vmalloc) [inline Makefile:8]: Won't this use Python outside of the venv?

Comment (ayalash) [reply Makefile:8]: No, UV use the virtualenv, even if you didn't activate it...

**Review by vmalloc — APPROVED** [review]

**Comment by vmalloc** [comment]
Approved. I will work on the 3.13 issue on top of this branch. Thanks!

**Comment by ayalash** [comment]
@vmalloc For python versions < 3.13 the `getfullargspec` raise `TypeError`:
```
python -c 'import time; import inspect; print(repr(inspect.getfullargspec(time.time)))'
```
For python version == 3.13 it will return:
```
FullArgSpec(args=[], varargs=None, varkw=None, defaults=None, kwonlyargs=[], kwonlydefaults=None, annotations={})
```

**Comment by ayalash** [comment]
Closing...
Will be handled as part of https://github.com/vmalloc/pyforge/pull/29

**Pull request closed without merging**